client/layout/guided-tours/config-elements/step.js

```diff
@@ -182,7 +182,7 @@ export default class Step extends Component {
 		this.setAnalyticsTimestamp( context );
 
 		if ( when && ! isValid( when ) ) {
-			const nextStepName = anyFrom( branching[ step ] );
+			const nextStepName = props.next || anyFrom( branching[ step ] );
```

Comment:
As per discussion on Slack we agree that we'd need to find a different name because `next` and `nextStepName` are taken and any alternative solution would be uglier still so we'll just go with this. 😄